code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -54,3 +54,108 @@
 	max_charges = 10
 	recharge_rate = 2
 	no_den_usage = 1
+
+
+// Lord Staves
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant
+	name = "staff of revenant"
+	desc = "A cursed artifact that starts off weak, but you can drain the souls of dead bodies in order to make it more powerful! Activate the staff in hand to see how many souls you have and, if you have enough, make your staff stronger."
+	fire_sound = "sound/magic/WandODeath.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/staffofrevenant
+	icon_state = "staffofrevenant"
+	item_state = "staffofrevenant"
+	w_class = 4
+	max_charges = 1
+	recharge_rate = 10
+	no_den_usage = 1
+	var/revenant_level = 0
+	var/revenant_damage = 20 // doesn't actually change the damage, only used for the inhand activation
+	var/revenant_souls = 0
+	var/list/drained_mobs = list() //Cannot harvest the same mob twice
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant/attack(mob/living/carbon/human/target, mob/living/user)
+	if(target.stat & DEAD)
```

Comment:
I would add an else after this that says "[target] is not dead, you can't drain their soul!"